packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -790,4 +849,113 @@ class PlatformViewRenderBox extends RenderBox {
     config.isSemanticBoundary = true;
     config.platformViewId = _controller.viewId;
   }
+
+  @override
+  void detach() {
+    _gestureRecognizer.reset();
+    super.detach();
+  }
+}
+
+// This recognizer constructs gesture recognizers from a set of gesture recognizer factories
+// it was given, adds all of them to a gesture arena team with the _PlatformViewGestureRecognizerCaptain
+// as the team captain.
+// As long as the gesture arena is unresolved the recognizer caches all pointer events.
+// When the team wins the recognizer sends all the cached point events to the [PlatformViewController], and
+// sets itself to a "forwarding mode" where it will forward any new pointer event to the [PlatformViewController].
+class _PlatformViewGestureRecognizerCaptain extends OneSequenceGestureRecognizer {
```

Comment:
This is almost identical to the existing _AndroidViewGestureRecognizer, the only difference is `dispatcher.handlePointerEvent` vs. `controller.dispatchPointerEvent`.

How about we make the original implementation take a `_PointerEventDispatcher` and use it for both RenderAndroidView and RenderPlatformView ?